### PR TITLE
checkboxes added for filtering of subscribed and requested channels

### PR DESCRIPTION
**Pull request by JuriBog**
# Description of change

In the channels overview the user can now search for subscribed and requested channels.

## Type of change

-   Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Manually. For that run the docker-compose file in integration-services from branch `25-query-subscribed-channels`.
Stop the is-dashboard container and run the dashboard from branch `38-search-for-subscribed-channels`.
Use a CORS plugin in the browser.

## Change checklist

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code

### CODE REVIEW

File: src/lib/app/streams.ts [hunk 1]
```diff
@@ -54,17 +56,19 @@ export function resetStreamsState(): void {
 // stored in the searchChannelsResults store.
 // TODO: Improve search algorithm, now it is searching only by author id or topic type
 let index = 0
-export async function searchAllChannels(query: string, options?: { limit?: number; authorId?: string }): Promise<void> {
+export async function searchAllChannels(query: string, options?: SearchOptions): Promise<void> {
     const _search = async (
         _searchAllHash: string,
         query: string,
-        options?: { limit?: number; authorId?: string }
+        options?: { limit?: number; authorId?: string; subscribedId?: string, requestedSubscriptionId?: string }
```
Comment (Schereo) [inline src/lib/app/streams.ts:63]: I think you can also use the `SearchOption` type here

File: src/lib/app/streams.ts [hunk 2]
```diff
@@ -97,15 +101,24 @@ export async function searchAllChannels(query: string, options?: { limit?: numbe
...
 export async function searchChannelsSingleRequest(
     query: string,
-    options: { searchByAuthorId?: boolean; searchBySource?: boolean; authorId?: string; limit: number; index?: number }
+    options: {
+        searchByAuthorId?: boolean;
+        searchBySource?: boolean;
+        authorId?: string;
+        subscribedId?: string,
+        requestedSubscriptionId?: string,
+        limit: number; index?: number
```
Comment (Schereo) [inline src/lib/app/streams.ts:110]: Weird formatting, please remove `;`

File: src/lib/app/streams.ts [hunk 3]
```diff
@@ -54,17 +56,19 @@ export function resetStreamsState(): void {
 // stored in the searchChannelsResults store.
 // TODO: Improve search algorithm, now it is searching only by author id or topic type
```
Comment (Schereo) [inline src/lib/app/streams.ts:57]:
```suggestion
```

File: src/lib/app/streams.ts [hunk 4]
```diff
@@ -9,17 +9,19 @@ import { AccessRights, type ChannelInfo } from '@iota/is-client'
 import { get } from 'svelte/store'
 import { authenticationData, channelClient, isAuthenticated } from './base'
 import { DEFAULT_SDK_CLIENT_REQUEST_LIMIT } from './constants/base'
-import { DEFAULT_AUTHOR_FILTER_STATE, FEED_INTERVAL_MS } from './constants/streams'
+import { DEFAULT_AUTHOR_FILTER_STATE, DEFAULT_REQUESTED_SUBSCRIPTION_STATE, DEFAULT_SUBSCRIBED_FILTER_STATE, FEED_INTERVAL_MS } from './constants/streams'
 import { showNotification } from './notification'
 import { NotificationType } from './types/notification'
-import { SubscriptionState } from './types/streams'
+import { SubscriptionState, type SearchOptions } from './types/streams'
```
[the remark this answers is not in the document]
Comment (dominic22) [reply src/lib/app/streams.ts:15]: yes I have but if I am the only one having this I'll try to fix it :D

File: src/lib/app/streams.ts [hunk 5]
```diff
@@ -456,3 +469,7 @@ export function isUserOwnerOfChannel(userDID: string, channel: ChannelInfo): boo
 export function isUserSubscribedToChannel(userDID: string, channel: ChannelInfo): boolean {
     return channel?.subscriberIds?.includes(userDID) && channel?.authorId !== userDID
 }
+
+export function hasUserRequestedSubscriptionToChannel(userDID: string, channel: ChannelInfo): boolean{
```
Comment (Schereo) [inline src/lib/app/streams.ts:473]: So we are not including own channels into subscribed channels?

Comment (JuriBog) [reply src/lib/app/streams.ts:473]: No. The owner of a channel we filter by the authorId. On creation of a channel a Subscription is created but not included in the `subscribedIds` list this is only done when subscribing to other channels where the user is not the owner of.

**Review by dominic22 — APPROVED** [review]

**Pull request merged**